hack/assets/providercustomizations.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func providerCustomizations(obj *unstructured.Unstructured, providerName string) {
+	switch providerName {
+	case "operator":
+		operatorCustomizations(obj)
+	case "azure":
+		azureCustomizations(obj)
+	case "gcp":
+		gcpCustomizations(obj)
+	}
+}
+
+func operatorCustomizations(obj *unstructured.Unstructured) {
+	switch obj.GetKind() {
+	case "Deployment":
+		deployment := &appsv1.Deployment{}
+		if err := scheme.Convert(obj, deployment, nil); err != nil {
+			panic(err)
```

Comment:
Why a panic? Is this always a programming error?

Reply:
Yes, this is something that should never happen